core/src/main/java/org/apache/iceberg/BaseMetadata.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg;
+
+import java.io.Serializable;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.apache.iceberg.util.PropertyUtil;
+import org.immutables.value.Value;
+
+/**
+ * A base class for {@link TableMetadata} and {@link org.apache.iceberg.view.ViewMetadata} It
+ * abstracts out all the common metadata for table ane view.
+ */
+public interface BaseMetadata extends Serializable {
+
+  @Value.Parameter(order = 1003)
+  String location();
+
+  @Value.Parameter(order = 1008)
+  Map<String, String> properties();
+
+  @Value.Parameter(order = 1010)
+  @Nullable
+  String metadataFileLocation();
+
+  @Value.Parameter(order = 1011)
+  Schema schema();
```

Comment:
Is this necessary to make this in the common base class, to allow ViewOperations  ?   Should we remove it , if it is not required?

Reply:
Yes , Schema is needed for Both table and view operations. Ref: https://github.com/apache/iceberg/pull/8907/files#diff-db46657b84d66e084e15f31b8dab21577efb2ae7102863f94c6c9477782de676R160